src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx

```diff
@@ -25,27 +25,30 @@ const getAvailableLanguages$ = (
   http: HttpSetup,
   data: DataPublicPluginSetup
 ) =>
-  data.query.dataSetManager.getUpdates$().pipe(
-    startWith(data.query.dataSetManager.getDataSet()),
-    distinctUntilChanged(),
-    switchMap(async (dataset) => {
-      // currently query assist tool relies on opensearch API to get index
-      // mappings, external data source types (e.g. s3) are not supported
-      if (dataset?.dataSource?.type !== DEFAULT_QUERY.ENGINE_TYPE) return [];
+  data.query.queryString
+    .getDatasetManager()
+    .getUpdates$()
+    .pipe(
+      startWith(data.query.queryString.getDatasetManager().getDataset()),
+      distinctUntilChanged(),
+      switchMap(async (dataset) => {
+        // currently query assist tool relies on opensearch API to get index
+        // mappings, external data source types (e.g. s3) are not supported
+        if (dataset?.dataSource?.type !== DEFAULT_QUERY.DATASET.DATASOURCE.TYPE) return [];
 
-      const dataSourceId = dataset?.dataSource?.id;
-      const cached = availableLanguagesByDataSource.get(dataSourceId);
-      if (cached !== undefined) return cached;
-      const languages = await http
-        .get<{ configuredLanguages: string[] }>(API.QUERY_ASSIST.LANGUAGES, {
-          query: { dataSourceId },
-        })
-        .then((response) => response.configuredLanguages)
-        .catch(() => []);
-      availableLanguagesByDataSource.set(dataSourceId, languages);
-      return languages;
-    })
-  );
+        const dataSourceId = dataset?.dataSource?.id;
+        const cached = availableLanguagesByDataSource.get(dataSourceId);
+        if (cached !== undefined) return cached;
+        const languages = await http
```

Comment:
Also not for this PR, but why are we making a HTTP call for this? Doesnt it simply need to check for `PPL` and whether its an `index` or `index-pattern`? All of which should be available in the the query object